server/athenian/api/internal/miners/jira/issue.py

```diff
@@ -87,14 +87,12 @@ async def generate_jira_prs_query(
         filters.append(_issue.type.in_(jira.issue_types))
     if jira.priorities:
         filters.append(_issue.priority_name.in_(jira.priorities))
+    filters.append(_issue.is_deleted.is_(False))
+
     if not jira.epics:
-        filters.extend(
-            [
-                _issue.is_deleted.is_(False),
-                _issue.acc_id == jira.account,
-                _issue.project_id.in_(jira.projects),
-            ],
-        )
+        filters.append(_issue.acc_id == jira.account)
+        if jira.projects:
+            filters.append(_issue.project_id.in_(jira.projects))
```

Comment:
```suggestion
        filters.append(_issue.project_id.in_(jira.projects))
```
I smell a potential business logic change. We must ALWAYS filter by allowed projects. The situation when `jira.projects` is empty is a big nasty assertion that we failed to load the allowed projects for the account. We must always load the allowed projects everywhere, this is super performance critical.

Reply:
reverted here.
`jira.projects` can still be empty, for instance an invalid project name is received from outside, or a project previously set in `TeamGoal.projects` becomes disabled